crates/walrus-sui/src/wallet.rs

```diff
@@ -0,0 +1,132 @@
+    }
+
+    // TODO: move callsites to the RetriableSuiClient.
+    /// Passes through to the `WalletContext` to get the reference gas price.
+    #[deprecated(note = "Avoid this method. Use the RetriableSuiClient instead.")]
+    pub async fn get_reference_gas_price(&self) -> Result<u64, WalletError> {
+        self.wallet_context.get_reference_gas_price().await
+    }
+
+    // TODO: move callsites to the RetriableSuiClient.
+    /// Passes through to the `WalletContext` to get an [`sui_types::base_types::ObjectRef`].
+    #[deprecated(note = "Avoid this method. Use the RetriableSuiClient instead.")]
+    pub async fn get_object_ref(&self, object_id: ObjectID) -> Result<ObjectRef, WalletError> {
+        self.wallet_context.get_object_ref(object_id).await
+    }
+
+    // TODO: move callsites to the RetriableSuiClient.
+    /// Passes through to the `WalletContext` to call execute_transaction_may_fail.
+    #[deprecated(note = "Avoid this method. Use the RetriableSuiClient instead.")]
+    pub async fn execute_transaction_may_fail(
+        &self,
+        signed_transaction: Envelope<SenderSignedData, EmptySignInfo>,
+    ) -> Result<SuiTransactionBlockResponse, WalletError> {
+        self.wallet_context
+            .execute_transaction_may_fail(signed_transaction)
+            .await
+    }
+
+    // TODO: move callsites to the RetriableSuiClient.
+    /// Find a gas object which fits the budget
+    #[deprecated(note = "Avoid this method. Use the RetriableSuiClient instead.")]
+    pub async fn gas_for_owner_budget(
+        &self,
+        address: SuiAddress,
+        budget: u64,
+        forbidden_gas_objects: BTreeSet<ObjectID>,
+    ) -> Result<(u64, SuiObjectData), WalletError> {
+        self.wallet_context
+            .gas_for_owner_budget(address, budget, forbidden_gas_objects)
+            .await
+    }
+
+    /// Update the `Move.lock` file with automated address management info. See
+    /// [`sui_package_management::update_lock_file`] for details.
+    pub async fn update_lock_file(
+        &self,
+        lock_command: LockCommand,
+        install_dir: Option<PathBuf>,
+        lock_file: Option<PathBuf>,
+        response: &SuiTransactionBlockResponse,
+    ) -> Result<(), WalletError> {
+        sui_package_management::update_lock_file(
+            &self.wallet_context,
+            lock_command,
+            install_dir,
+            lock_file,
+            response,
+        )
+        .await
+    }
+
+    /// Get the rpc_url for the active environment.
+    #[deprecated(
+        note = "Avoid calling this method as it relies on the Wallet configuration for rpc urls. \
+        It's better to pass rpc urls directly to the location where they are needed, from Walrus \
+        configuration."
+    )]
+    pub fn get_rpc_url(&self) -> Result<String, WalletError> {
+        Ok(self.wallet_context.config.get_active_env()?.rpc.clone())
+    }
+    /// Get the path to the wallet configuration file.
```

Comment:
Nit: Can you add blank lines between functions?

(Side note: I'm surprised that rustfmt doesn't actually do this... 🤔)

Reply:
Done.